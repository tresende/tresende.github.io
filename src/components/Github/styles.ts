import styled from 'styled-components'

export const Wrapper = styled.div`
  @media only screen and (max-width: 760px) {
    .github-corner {
      display: none;
    }
  }
`
