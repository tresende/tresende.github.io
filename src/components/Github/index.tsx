import GithubCorner from 'react-github-corner'

import * as S from './styles'

const Github = () => (
  <S.Wrapper>
    <GithubCorner bannerColor="greenyellow" octoColor="#000" href="https://github.com/tresende/tresende.github.io" />
  </S.Wrapper>
)

export default Github
