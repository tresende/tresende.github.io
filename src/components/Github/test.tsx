import { render, screen } from '@testing-library/react'

import Github from '.'
const GITHUB_URL = 'https://github.com/tresende/tresende.github.io'

describe('<Github />', () => {
  it('should render with link', () => {
    const { container } = render(<Github />)

    expect(screen.getByRole('link')).toHaveAttribute('href', GITHUB_URL)
    expect(container.firstChild).toMatchSnapshot()
  })
})
