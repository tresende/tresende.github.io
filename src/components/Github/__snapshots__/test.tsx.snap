// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Github /> should render with link 1`] = `
@media only screen and (max-width:760px) {
  .c0 .github-corner {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <a
      aria-label="Open GitHub project"
      class="github-corner"
      href="https://github.com/tresende/tresende.github.io"
    >
      <svg
        height="80"
        style="position: absolute; top: 0px; fill: #000; right: 0px;"
        viewBox="0 0 250 250"
        width="80"
      >
        <path
          class="octo-banner"
          d="M0 0l115 115h15l12 27 108 108V0z"
          fill="greenyellow"
        />
        <path
          class="octo-arm"
          d="M128 109c-15-9-9-19-9-19 3-7 2-11 2-11-1-7 3-2 3-2 4 5 2 11 2 11-3 10 5 15 9 16"
          style="transform-origin: 130px 106px;"
        />
        <path
          class="octo-body"
          d="M115 115s4 2 5 0l14-14c3-2 6-3 8-3-8-11-15-24 2-41 5-5 10-7 16-7 1-2 3-7 12-11 0 0 5 3 7 16 4 2 8 5 12 9s7 8 9 12c14 3 17 7 17 7-4 8-9 11-11 11 0 6-2 11-7 16-16 16-30 10-41 2 0 3-1 7-5 11l-12 11c-1 1 1 5 1 5z"
        />
      </svg>
    </a>
  </div>
</div>
`;
